repair/featurize/constraintfeat.py

```diff
@@ -46,30 +45,50 @@
                               '                AND t3.rv_val $operation $rv_val)')
 
 
-def gen_feat_tensor(violations, total_vars, classes):
-    tensor = torch.zeros(total_vars,classes,1)
-    if violations:
-        for entry in violations:
-            vid = int(entry[0])
-            val_id = int(entry[1]) - 1
-            feat_val = float(entry[2])
-            tensor[vid][val_id][0] = feat_val
-    return tensor
-
-
 class ConstraintFeaturizer(Featurizer):
     def specific_setup(self):
         self.name = 'ConstraintFeaturizer'
         self.constraints = self.ds.constraints
         self.init_table_name = self.ds.raw_data.name
+        self.featurization_query_results = self._get_featurization_query_results()
+        # List[Dict[str, Dict[str, int]]] where the key of the dict is the vid
+        # and the value is a dict whtere the key is the value_id and the value
+        # is the number of violations.
+        self.featurization_maps = self.build_featurization_maps(self.featurization_query_results)
+
+    # TODO(jmio): potential memory issues
+    def build_featurization_maps(self, queries):
```

Comment:
Can you add a docstring here specifying the structure of `queries`? Also above `self.featurization_query_results`.